devito/equation.py

```diff
@@ -80,8 +80,16 @@ def solve(eq, target, **kwargs):
                    the equation. For more information. refer to
                    ``sympy.solve.__doc__``.
     """
+    from devito.finite_differences.differentiable import Differentiable
     # Enforce certain parameters to values that are known to guarantee a quick
     # turnaround time
     kwargs['rational'] = False  # Avoid float indices
     kwargs['simplify'] = False  # Do not attempt premature optimisation
-    return sympy.solve(eq, target, **kwargs)[0]
+    # Reset to eqn = 0 if provided an Eq object
+    if isinstance(eq, Eq):
+        eq = eq.lhs - eq.rhs
+
+    if isinstance(eq, Differentiable):
```

Comment:
do you actually need this ``if``? even though ``Differentiable``'s ``func`` is defined? that's weird